modules/solid_mechanics/src/timeintegrators/CentralDifferenceDirect.C

```diff
@@ -0,0 +1,194 @@
+
+  // Compute the residual
+  _explicit_residual.zero();
+  _fe_problem.computeResidual(
+      *_nonlinear_implicit_system->current_local_solution, _explicit_residual, _nl.number());
+
+  // Move the residual to the RHS
+  _explicit_residual *= -1.0;
+
+  auto & mass_matrix = _nonlinear_implicit_system->get_system_matrix();
+
+  // Compute the mass matrix
+  if (!_constant_mass || (_constant_mass && _t_step == 1))
+    _fe_problem.computeJacobianTag(
+        *_nonlinear_implicit_system->current_local_solution, mass_matrix, mass_tag);
+
+  // Perform the linear solve
+  bool converged = performExplicitSolve(mass_matrix);
+
+  // Update the solution
+  *_nonlinear_implicit_system->solution = _nl.solutionOld();
+  *_nonlinear_implicit_system->solution += _solution_update;
+
+  _nonlinear_implicit_system->nonlinear_solver->converged = converged;
+
+  // Applying preset BCs
+  setPresetBCs();
+}
+
+void
+CentralDifferenceDirect::setPresetBCs()
+{
+  // Apply preset nodal BC's
+  NumericVector<Number> & initial_solution(*_nonlinear_implicit_system->solution);
+
+  auto & nodal_bcs = _nl.getPresetBCWarehouse();
+  const ConstBndNodeRange & bnd_nodes = _fe_problem.getCurrentAlgebraicBndNodeRange();
+  for (const auto & bnode : bnd_nodes)
+  {
+    BoundaryID boundary_id = bnode->_bnd_id;
+    Node * node = bnode->_node;
+    if (node->processor_id() == processor_id())
+    {
+      // reinit variables in nodes
+      _fe_problem.reinitNodeFace(node, boundary_id, 0);
+
+      if (nodal_bcs.hasActiveBoundaryObjects(boundary_id))
+      {
+        const auto & preset_bcs = nodal_bcs.getActiveBoundaryObjects(boundary_id);
+        for (const auto & preset_bc : preset_bcs)
+          preset_bc->computeValue(initial_solution);
+      }
+      if (nodal_bcs.hasActiveBoundaryObjects(boundary_id))
+      {
+        const auto & preset_bcs_res = nodal_bcs.getActiveBoundaryObjects(boundary_id);
+        for (const auto & preset_bc : preset_bcs_res)
+          preset_bc->computeValue(initial_solution);
+      }
+    }
+  }
+}
+
+void
+CentralDifferenceDirect::postResidual(NumericVector<Number> & residual)
+{
+  residual += _Re_time;
+  residual += _Re_non_time;
+  residual.close();
+
+  // Reset time to the time at which to evaluate nodal BCs, which comes next
+  _fe_problem.time() = _current_time;
+}
+
+bool
+CentralDifferenceDirect::performExplicitSolve(SparseMatrix<Number> & mass_matrix)
+{
+  bool converged = false;
+
+  // Computes the sum of each row (lumping)
+  // Note: This is actually how PETSc does it
+  // It's not "perfectly optimal" - but it will be fast (and universal)
+  mass_matrix.vector_mult(_mass_matrix_diag, *_ones);
+
+  // "Invert" the diagonal mass matrix
+  _mass_matrix_diag.reciprocal();
+
+  // Calculate acceleration
+  auto & accel = *_sys.solutionUDotDot();
+  accel.pointwise_mult(_mass_matrix_diag, _explicit_residual);
+
+  // Scaling the acceleration
+  auto accel_scaled = accel.clone();
+  accel_scaled->scale((_dt + _dt_old) / 2);
```

Comment:
This is different from the math you showed in the PR description.

Reply:
Good catch, I need to update the documentation. This is the correct version

Reply:
Okay, then this is different from the original paper you referenced. I think this will require some documentation (i.e. why (dt+dt_old)/2 instead of dt).

Reply:
I'm guessing the paper I referenced assumed constant dt. The way I've implemented does not

Reply:
Why did you choose (dt+dt_old)/2? That doesn't seem to follow the constant acceleration assumption.

Reply:
Check my math, assuming constant acceleration and linear velocity
```
v_{n+1/2} = v_{n-1/2} + dt_old / 2 * a_{n-1} + dt / 2 * a_n
```
Your implementation assumes
```
(dt_old+dt) / 2 * a_n = dt_old / 2 * a_{n-1} + dt / 2 * a_n
```
which is questionable.

Reply:
The first statement should be written as 
`v_{n+1/2} = v_{n-1/2} + dt_old / 2 * a_n + dt / 2 * a_n`

Really this is just defining an effective time step to jump between mid points when the time step is not constant. 
`dt_eff = (dt_{n} + dt_{n+1})/2`
`v_{n+1/2} = v_{n-1/2} +  dt_eff  * a_n`

[Abaqus uses the same method. ](https://classes.engineering.wustl.edu/2009/spring/mase5513/abaqus/docs/v6.6/books/gsx/default.htm?startat=ch03s02.html)

If `t_{n-1} = 1`, `t_{n}=2` and `t_{n+1} = 2.5`, we want `t_{n-1/2}+dt_eff = 2.25 = t_{n+1/2}`
In this case `dt_eff=0.75`, so this would return the correct value for`t_{n+1/2}`

Reply:
Okay, I see. So this assumes constant acceleration between midpoints, not between time step begin and end. Your last comment can be a good documentation. Make sure you also reference the Abaqus manual. I'll resolve this conversation once I see the relevant documentation added.

Reply:
[f4073ac](https://github.com/idaholab/moose/pull/28175/commits/f4073ace5c7337485b751e904d6d55d78ed7b604)

I've updated the documentation here